### ISSUE TITLE
Deploy on zkSync2.0

### ISSUE THREAD

**Issue by open-contracts** [cartesi/machine-solidity-step#35]
It would be so cool to have a ZKP about a Linux machine. Which basically just means deploying this contract to the [zkSync2.0 testnet](https://blog.matter-labs.io/zksync-2-0-public-testnet-is-live-de870ba9632a). How long would it take to generate a proof for the execution of a simple python script?

**Comment by augustoteixeira** [comment]
Hi there, thanks for the message!

We have been considering a ZK version of the Cartesi consensus, although still in the speculative stage.

This video gives a good explanation of a possible path forward:

https://www.youtube.com/watch?v=6wLSkpIHXM8

This would involve several steps:
- getting more familiarity with some of ZkSync's code base
- compiling RISC-V to TinyRAM
- understanding how to optimize the recursive aggregation for this context.

The benefits of this would be amazing, including faster finalization and a better trust model.

However, we are quite scared about a possible bad performance of such a solution. We need to improve our research in order to answer questions like the one you posed (on the time needed to generate proofs).

So, in short: this is in our radar, but it is still speculative and we are focusing more on delivering an end-to-end solution for Linux decentralized applications before considering moving to ZK, which we see as a nice but complex addition.
